src/tax-lot/tax-lot.repo.ts

```diff
@@ -0,0 +1,175 @@
+import { Inject } from "@nestjs/common";
+import { DB, DbType } from "src/global/providers/db.provider";
+import { eq, sql } from "drizzle-orm";
+import { DataRetrievalException, ResourceNotFoundException } from "src/error";
+import {
+  SelectTaxLot,
+  SelectTaxLotNested,
+  SelectTaxLotSpatial,
+} from "src/schema/tax-lot";
+import {
+  taxLot,
+  zoningDistrict,
+  zoningDistrictClass,
+  zoningDistrictZoningDistrictClass,
+} from "src/schema";
+import { MultiPolygon } from "geojson";
+
+export class TaxLotRepo {
+  constructor(
+    @Inject(DB)
+    private readonly db: DbType,
+  ) {}
+
+  #checkTaxLotByBbl = this.db.query.taxLot
+    .findFirst({
+      columns: {
+        bbl: true,
+      },
+      where: (taxLot, { eq, sql }) => eq(taxLot.bbl, sql.placeholder("bbl")),
+    })
+    .prepare("checkTaxLotByBbl");
+
+  async taxLotByBbl(bbl: string) {
+    let result: SelectTaxLotNested | undefined;
+    try {
+      result = await this.db.query.taxLot.findFirst({
+        columns: {
+          boroughId: false,
+          landUseId: false,
+          wgs84: false,
+          liFt: false,
+        },
+        where: (taxLot, { eq }) => eq(taxLot.bbl, bbl),
+        with: {
+          borough: true,
+          landUse: true,
+        },
+      });
+    } catch {
+      throw DataRetrievalException;
+    }
+
+    if (result === undefined) throw ResourceNotFoundException;
+    return result;
+  }
+
+  async taxLotByBblGeoJson(bbl: string) {
+    let result: SelectTaxLotSpatial | undefined;
+    try {
+      result = await this.db.query.taxLot.findFirst({
+        columns: {
+          bbl: true,
+          block: true,
+          lot: true,
+          address: true,
+        },
+        extras: {
+          geometry: sql<string>`ST_AsGeoJSON(${taxLot.wgs84}, 6)`.as(
+            "geometry",
+          ),
+        },
+        where: (taxLot, { eq }) => eq(taxLot.bbl, bbl),
+        with: {
+          borough: true,
+          landUse: true,
+        },
+      });
+    } catch {
+      throw DataRetrievalException;
+    }
+
+    if (result === undefined) throw ResourceNotFoundException;
+
+    const geometry = JSON.parse(result.geometry) as MultiPolygon;
+
+    return {
+      type: "Feature",
+      id: result.bbl,
+      properties: {
+        bbl: result.bbl,
+        borough: result.borough,
+        block: result.block,
+        lot: result.lot,
+        address: result.address,
+        landUse: result.landUse,
+      },
+      geometry,
+    };
+  }
+
+  async zoningDistrictByTaxLotBbl(bbl: string) {
```

Comment:
Some feedback about method naming that I would love @TangoYankee's two cents on as well. I'm leaving this comment here as an example but this applies generally to what we name methods in services as well as repositories:

* I think we should have a convention of having some kind of verb in the name. So `zoningDistrictByTaxLotBbl` would be something like `findZoningDistrictByTaxLotBbl`. 
* I think we should avoid including words in method and variable names that are redundant in the context of where the method/variable is defined. In the case of services and repositories, this usually means taking out words that are already in the class name of the class that the method belongs too. For example, if you look at where this method is being called, we do `this.taxLotRepo.zoningDistrictByTaxLotBbl(bbl);`. A good way to keep our method names from being too huge would be to simplify this to `zoningDistrictByBbl`, so you would do `this.taxLotRepo.zoningDistrictByBbl(bbl);`. This is still a good name because `taxLotRepo` tells you that you're dealing with taxLots.
* One other minor nit pick - this should probably be `zoningDistricts` not `zoningDistrict`, because it gives back a list of districts.

Putting these suggestions together, this method name might be something like `findZoningDistrictsByBbl`.

Reply:
These conventions make sense to me. I also like that the class names remain singular in these conventions. In my mind, classes map to database tables, which are singular. 

In keeping with the general ticket workflow, only the method names that were created in this ticket should be updated as part of this feedback. This means there are several pre-existing classes with unexamined method names that should be scrutinized and refactored. I'll create follow-up tickets to refactor the methods on other classes.

Reply:
Makes sense @TylerMatteo . I will update the names. Regarding the nitpick. Can a BBL have more than one zoning district? This particular method  `findZoningDistrictByTaxLotBbl` only returns a singular zoning district.

Reply:
> This particular method findZoningDistrictByTaxLotBbl only returns a singular zoning district.

That is the intuitive understanding- each tax lot is associated with only one zoning district. However, zoning districts are not required to follow tax lot boundaries. This leads to situations where multiple zoning districts may intersect a tax lot (or a tax lot may not be inside any zoning district!)

Reply:
Makes sense, I'll update the names. 

Regarding the nitpick, can a BBL be in more than one zoning district? AFAIK, this particular method `findZoningDistrictClassByBbl` only returns a singular zoning district.

Reply:
> can a BBL be in more than one zoning district?

Yes, a bbl may have multiple zoning districts that intersect it. It may also be helpful to consider the difference between 'being in' and 'intersecting'. It may be natural to think of 'being in' as 'the whole tax lot is completely within the zoning district.' However, we are asking whether '**parts** of the tax lot overlap with a zoning district'. 

Because of the way zoning districts are drawn, one part of a tax lot may be in zoning district 'A' while another part of that tax lot is in zoning district 'B'. Check out `1014400022` [locally](http://localhost:3000/api/tax-lots/1014400022/zoning-districts) or on [Zola](https://zola.planning.nyc.gov/l/lot/1/1440/22#18.85/40.763761/-73.9589043) . This bbl is in two zoning districts - a `C1-9` and an `R8B`